CenterForOpenScience/osf.io
# [RTM][ENG-2448] add new landing pages for sso-migration institutions

## Purpose

Provide a migration path for users from institutions that no longer support institutional login.

Corresponding [newCAS](https://github.com/CenterForOpenScience/osf-cas/pull/14) & [oldCAS](https://github.com/CenterForOpenScience/cas-overlay/pull/195) tickets create pages that allow the user to request a password reset email if their institution is no longer supported.  This PR adds the necessary osf-side changes to support that.

## Changes

When an existing user from a recently-disabled institution enters OSF, present them with a bespoke page to allow them to set a new, OSF-specific password.  These pages reuse the existing OSF password reset functionality but provide extra context for institutional users in the displayed messages.

## QA Notes

This needs to be tested in combination with either the newCAS or oldCAS PRs.

The expected login workflow is:

* Institutional user clicks "Sign In" button on the OSF frontpage
* They click "Sign in via Institution" button
* Their institution isn't present in the "Select Institution" drop-down, so they click "Can't find my institution"
* They do not have an ORCiD-created account or a previously-OSF-created account.
* They type in their email address into the form.
* They are taken to a page that confirms a "reset password" email has been sent to the provided email.
* If they are the owner of the provided email address, they receive an email that allows them to create an OSF-specific password.

### What are the areas of risk?

Overlooked endpoints. When developing, I assumed that the only access to the "forgot-password" page would be through CAS.  When accessing it directly, I encountered a 500.  That has since been fixed, but could use some critical evaluation

### Any concerns/considerations/questions that development raised?

I don't think so?

## Documentation

This is a very specific endpoint and probably(?) does not need documentation beyond what is provided in the code.

## Side Effects

None expected.

## Ticket

oldCAS: https://openscience.atlassian.net/browse/ENG-2241
newCAS: https://openscience.atlassian.net/browse/ENG-2448
